dataset.py

```diff
@@ -20,6 +21,23 @@
 
 # global variables
 w2i = None
+i2w = None
+
+
+# take the sequence of outputs from the transformer and convert it into a sentence
+# assumes w2i was already computed
+def translate_output(transformer_outputs: List[torch.Tensor]) -> str:
+    global w2i
+    global i2w
+    if w2i is None or i2w is None:
+        fill_w2i()
+    result = ''
+    for output in transformer_outputs:
+        i = np.argmax(output)
```

Comment:
Don't mix np with torch. Maybe pass np.ndarray as input to the function instead of tensor?

Reply:
just used torch argmax instead

Reply:
good catch, kar655, u never disappoint